src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -161,25 +168,63 @@ internal static BindingDataResolver GetAppSettingResolver(string originalValue,
             return (newAttr, bindingData) => resolvedValue;
         }
 
+        // Run validition. This needs to be run at different stages. 
+        // In general, run as early as possible. If there are { } tokens, then we can't run until runtime.
+        // But if there are no { }, we can run statically. 
+        // If there's no [AutoResolve], [AppSettings], then we can run immediately.
+        private static void RunValidateAttributes(object value, PropertyInfo propInfo)
+        {
+            var attrs = propInfo.GetCustomAttributes<ValidationAttribute>();
+
+            foreach (var attr in attrs)
+            {
+                try
+                {
+                    attr.Validate(value, propInfo.Name);
+                }
+                catch (Exception e)
+                {
+                    throw new InvalidOperationException($"'{propInfo.Name}' can't be '{value}': {e.Message}");
+                }
+            }
+        }
+
         // Resolve for AutoResolve.Default templates. 
         // These only have access to the {sys} builtin variable and don't get access to trigger binding data. 
-        internal static BindingDataResolver GetBuiltinTemplateResolver(string originalValue, INameResolver nameResolver)
+        internal static BindingDataResolver GetBuiltinTemplateResolver(string originalValue, INameResolver nameResolver, PropertyInfo propInfo)
         {
             string resolvedValue = nameResolver.ResolveWholeString(originalValue);
 
             var template = BindingTemplate.FromString(resolvedValue);
+            if (!template.ParameterNames.Any())
+            {
+                // No { } tokens, bind eagerly up front. 
+                RunValidateAttributes(originalValue, propInfo);
+            }
 
             SystemBindingData.ValidateStaticContract(template);
 
             // For static default contracts, we only have access to the built in binding data. 
-            return (newAttr, bindingData) => template.Bind(SystemBindingData.GetSystemBindingData(bindingData));
+            return (newAttr, bindingData) =>
+            {
+                var newValue = template.Bind(SystemBindingData.GetSystemBindingData(bindingData));
+                RunValidateAttributes(newValue, propInfo);
+                return newValue;
+            };
         }
 
         // AutoResolve
         internal static BindingDataResolver GetTemplateResolver(string originalValue, AutoResolveAttribute attr, INameResolver nameResolver, PropertyInfo propInfo, BindingDataContract contract)
```

Comment:
Ah nvmd, missed a line break in the github viewer, nvmd it's not clear duplication.